students/david_russo/lesson5/mailroom_lesson05.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+from sys import exit
+import random
+
+# construct list of donors with a frozen set, to later be used as keys in the dictionary
+donor_names = frozenset(("Daniel Ocean", "Rusty Ryan", "Linus Caldwell", "Ruben Tishkoff", "Basher Tarr"))
+
+# create donors with dict comprehension 
+donors = {donor: random.sample(range(1, 50000), random.randint(1, 5)) for donor in donor_names}
+
+# define a function to prompt the user for their desired action
+def prompt_user():
+    print("Choose an action. ")
+    print()
+    print("1 - Send a Thank You")
+    print("2 - Create a Report")
+    print("3 - Send letters to everyone")
+    print("4 - Quit")
+    # I previously had a 'while response not in (1, 2, 3, 4)' conditional which felt like a better choice than 
+    # this try except block. Should I put the try except into a while statement so that it continues to prompt
+    # the user for the correct type of response? Or is it good practice to error out if the user continues to 
+    # provide bad input? 
```

Comment:
Yes you should do this.  So then when you catch the except you can just put a `continue` and it will go the the top of the loop and ask again